tests/core/test_proxy.py

```diff
@@ -44,3 +65,63 @@ def test_no_proxy(self):
         env.pop("https_proxy", None)
         env.pop("HTTP_PROXY", None)
         env.pop("HTTPS_PROXY", None)
+
+
+class CustomAgentTransaction(AgentTransaction):
+
+    def on_response(self, response):
+        super(CustomAgentTransaction, self).on_response(response)
+        self._test.stop()
```

Comment:
Actually that's what I did at first, but then when the `on_response` method is called, the attribute `_test` got deleted somehow and I got `'CustomAgentTransaction' object has no attribute '_test'`. I didn't find what was wrong so I changed my approach.